src/libmongoc/tests/test-mongoc-secure-channel.c

```diff
@@ -0,0 +1,263 @@
+#include <mongoc/mongoc.h>
+
+#include "TestSuite.h"
+#include "test-libmongoc.h"
+
+#if defined(MONGOC_ENABLE_SSL_SECURE_CHANNEL)
+#include <mongoc/mongoc-stream-tls-secure-channel-private.h>
+#include <mongoc/mongoc-host-list-private.h> // _mongoc_host_list_from_string_with_err
+#include <mongoc/mongoc-client-private.h>    // mongoc_client_connect_tcp
+#include <mongoc/mongoc-log-private.h>       // _mongoc_log_get_handler
+#include <test-conveniences.h>
+
+
+static bool
+connect_with_secure_channel_cred (mongoc_ssl_opt_t *ssl_opt, mongoc_shared_ptr cred_ptr, bson_error_t *error)
+{
+   mongoc_host_list_t host;
+   const int32_t connect_timout_ms = 10000;
+
+   *error = (bson_error_t) {0};
+
+   if (!_mongoc_host_list_from_string_with_err (&host, "localhost:27017", error)) {
```

Comment:
Note: This line will break with #2047 because it changes the signature of this function. Whichever is merged first will require changing the other.